L1Trigger/TrackFindingTMTT/src/TP.cc

```diff
@@ -0,0 +1,189 @@
+  }
+
+  //=== Check if this tracking particle is worth keeping.
+  //=== (i.e. If there is the slightest chance of reconstructing it, so as to measure fake rate).
+
+  void TP::fillUse() {
+    const bool useOnlyInTimeParticles = false;
+    const bool useOnlyTPfromPhysicsCollisionFalse = false;
+    // Use looser cuts here those those used for tracking efficiency measurement.
+    // Keep only those TP that have a chance (allowing for finite track resolution) of being reconstructed as L1 tracks. L1 tracks not matching these TP will be defined as fake.
+
+    const vector<int> genPdgIdsAllUnsigned = {
+        11,
+        13,
+        211,
+        321,
+        2212};  // Include all possible particle types here, as if some are left out, L1 tracks matching one of missing types will be declared fake.
+    vector<int> genPdgIdsAll;
+    for (unsigned int i = 0; i < genPdgIdsAllUnsigned.size(); i++) {
+      genPdgIdsAll.push_back(genPdgIdsAllUnsigned[i]);
+      genPdgIdsAll.push_back(-genPdgIdsAllUnsigned[i]);
+    }
+
+    // Range big enough to include all TP needed to measure tracking efficiency
+    // and big enough to include any TP that might be reconstructed for fake rate measurement.
+    const float ptMin = min(settings_->genMinPt(), 0.7 * settings_->houghMinPt());
+    const float etaMax = max(settings_->genMaxAbsEta(), 0.2 + fabs(settings_->etaRegions()[0]));
+
+    static thread_local TrackingParticleSelector trackingParticleSelector(ptMin,
+                                                                          9999999999,
+                                                                          -etaMax,
+                                                                          etaMax,
+                                                                          max(10.0, settings_->genMaxVertR()),
+                                                                          max(35.0, settings_->genMaxVertZ()),
+                                                                          0,
+                                                                          useOnlyTPfromPhysicsCollisionFalse,
+                                                                          useOnlyInTimeParticles,
+                                                                          true,
+                                                                          false,
+                                                                          genPdgIdsAll);
+
+    const TrackingParticlePtr tp_ptr(*this);  // cast to base class.
+    use_ = trackingParticleSelector(*tp_ptr);
+  }
+
+  //=== Check if this tracking particle can be used to measure the L1 tracking efficiency.
+
+  void TP::fillUseForEff() {
+    useForEff_ = false;
+    if (use_) {
+      const bool useOnlyInTimeParticles = true;
+      const bool useOnlyTPfromPhysicsCollision = true;
+      static thread_local TrackingParticleSelector trackingParticleSelector(settings_->genMinPt(),
```

Comment:
Replaced by "static const"

Reply:
Is any of the `TrackingParticleSelector` constructor parameters coming from configuration (the use of `Settings` would hint towards that)? If yes, I advice against `static` because if anyone would use a different instance of `Settings` with different values, a `static const trackingParticeSelector` is initialized with one of the settings (whichever gets "first"), which may lead to surprising results (that may be subtle or not).